prov/efa/src/efa_prov_info.c

```diff
@@ -102,7 +102,7 @@ int efa_prov_info_set_fabric_attr(struct fi_info *prov_info, struct efa_device *
  */
 const struct fi_domain_attr efa_domain_attr = {
 	.caps			= EFA_DOMAIN_CAPS,
-	.threading		= FI_THREAD_DOMAIN,
+	.threading		= FI_THREAD_DOMAIN_RELAXED,
```

Comment:
We shouldn't change it here. Two reasons

1. For rdm ep type, it won't make any impact. efa finally changes it to FI_THREAD_SAFE as the default provider threading model: https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/efa_prov_info.c#L583-L587
2. Every provider must support FI_THREAD_SAFE. If we make it default to use FI_THREAD_DOMAIN, and application requests FI_THREAD_SAFE, this call will fail https://github.com/ofiwg/libfabric/blob/main/prov/util/src/util_attr.c#L563-L567